pkg/solidity-utils/test/BasePoolMath.test.ts

```diff
@@ -0,0 +1,199 @@
+import { bn, decimal, toFp } from '@balancer-labs/v3-helpers/src/numbers';
+import { MAX_RELATIVE_ERROR } from '@balancer-labs/v3-helpers/src/constants';
+import { deploy } from '@balancer-labs/v3-helpers/src/contract';
+import { expectEqualWithError } from '@balancer-labs/v3-helpers/src/test/relativeError';
+import { sharedBeforeEach } from '@balancer-labs/v3-common/sharedBeforeEach';
+import '@balancer-labs/v3-common/setupTests';
+import { expect } from 'chai';
+
+import {
+  computeProportionalAmountsIn,
+  computeProportionalAmountsOut,
+  computeAddLiquidityUnbalanced,
+  computeAddLiquiditySingleTokenExactOut,
+  computeRemoveLiquiditySingleTokenExactOut,
+  computeRemoveLiquiditySingleTokenExactIn,
+} from '@balancer-labs/v3-helpers/src/math/base';
+
+import { BasePoolMathMock } from '../typechain-types/contracts/test/BasePoolMathMock';
+
+const SWAP_FEE = decimal(0.01);
+
+describe('BasePoolMath', function () {
+  let math: BasePoolMathMock;
+
+  sharedBeforeEach('deploy', async function () {
+    math = await deploy('BasePoolMathMock');
+  });
+
+  it('test computeProportionalAmountsIn', async () => {
+    const balances = [decimal(100), decimal(200)];
+    const bptTotalSupply = decimal(300);
+    const bptAmountOut = decimal(30);
+
+    const expected = computeProportionalAmountsIn(balances, bptTotalSupply, bptAmountOut);
+    const result = await math.computeProportionalAmountsIn(
+      balances.map((balance) => bn(toFp(balance))),
+      bn(toFp(bptTotalSupply)),
```

Comment:
If we need to use this `decimal` type, maybe make new helpers in numbers.ts so that you don't have to chain these functions everywhere? Like `toBn` or `toFpAsBn`